tests/e2e/json_writer.go

```diff
@@ -0,0 +1,30 @@
+package main
+
+import (
+	"encoding/json"
+	"os"
+)
+
+// TraceWriter is an interface for writers that write steps to files.
+type TraceWriter interface {
+	WriteTraceToFile(filepath string, trace []Step) error
+}
+
+// JSONWriter is a simple writer that simply marshals the array of Step objects.
```

Comment:
nit: either "simple writer than marshals" or "a write that simply marshals"

Reply:
Thanks, 720962834ecb08ed3c124c8c4b1ce5dcb0cc8d5c